code/workspaces/auth-service/src/dataaccess/userRolesRepository.js

```diff
@@ -56,12 +56,23 @@ async function getUsers() {
   return Object.values(usersMap); // return users
 }
 
+function combineRoles(roles1, roles2) {
+  return {
+    ...roles1,
+    ...roles2,
+    projectRoles: [
+      ...(roles1 ? roles1.projectRoles : []),
+      ...(roles2 ? roles2.projectRoles : []),
+    ],
+  };
+}
+
 async function getAllUsersAndRoles() {
   const allRoles = await UserRoles().find().exec();
   const usersMap = allRoles
     .filter(roles => roles.userName) // only take users with known user names
     .reduce((uniqueUsersMap, roles) => { // convert to map, keyed by userId
-      uniqueUsersMap[roles.userId] = addDefaults(roles); // eslint-disable-line no-param-reassign
+      uniqueUsersMap[roles.userId] = combineRoles(uniqueUsersMap[roles.userId], addDefaults(roles)); // eslint-disable-line no-param-reassign
```

Comment:
One thing we might also need to be careful of is if for some reason the duplicated user has conflicting roles e.g. one says they are not an instance admin and the second says they are. In this case, the permission that is found second will be taken as the final permission. This might be something that will be rectified by the technical debt ticket you mentioned you have added.

Reply:
Having thought about this, combineRoles defaults to the first user, since that's the one most likely to be used by findOne.  Comment added.